src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -118,18 +110,18 @@ private void GetTypeMismatchDiagnostics(IDiagnosticWriter diagnosticWriter, IEnu
         public ImmutableArray<IDiagnostic> GetAllDiagnostics()
             => BicepParamFile.ProgramSyntax.GetParseDiagnostics()
                 .Concat(ParamsTypeManager.GetAllDiagnostics())
-                .Concat(this.compilationLoadDiagnostics)
+                .Concat(this.CompilationLoadDiagnostics)
                 .Concat(GetAdditionalSemanticDiagnostics()).ToImmutableArray();
             
         /// <summary>
         /// Gets the file that was compiled.
         /// </summary>
         public ParamFileSymbol Root => this.ParamBinder.ParamFileSymbol;
 
-        private Uri? TryGetBicepFileUri(out ToListDiagnosticWriter allDiagnostics)
+        public static Uri? TryGetBicepFileUri(out ToListDiagnosticWriter allDiagnostics, IFileResolver fileResolver, BicepParamFile bicepParamFile)
```

Comment:
Any thoughts on making this function also follow the double output pattern as ```TryGetUsingPath```?

Reply:
I don't think that signature is necessary here, since it's possible to get both a URI *and* diagnostics. The double out param + boolean return pattern is most useful when you can either get one output or the other.

But `ToListDiagnosticWriter` is not the right type for an output here, since the caller will not write to this once the `TryGetBicepFileUri` invocation is complete; they are instead expected to call `.GetDiagnostics()`. The method could take an IDiagnosticWriter as a parameter (meaning the caller would need to call `ToListDiagnosticWriter.Create()`) or output a collection of diagnostics (i.e., `out IEnumerable<IDiagnostic>`). I would probably choose the latter just because it has fewer moving parts from the caller's perspective.